### ISSUE TITLE
high auroc values and consistently 0 AP

### ISSUE THREAD

**Issue by jakubMitura14** [DIAGNijmegen/picai_eval#8]
Hello I am writing here becouse I do not have an idea why during training I get increasing auroc but constant 0 AP. I evaluate the model every 9 epochs on the validation part of the dataset.

below shortened example with single batch How I use evaluate method
```
#first I extract the lesion candidates
y_det=extract_lesion_candidates( y_det.cpu().detach().numpy()[1,:,:,:])[0]
#then I save y_det and y_true in files in given folder
```

on the validation end
```
valid_metrics = evaluate(y_det=self.list_yHat_val,
                                y_true=self.list_gold_val,
                                num_parallel_calls= os.cpu_count()

meanPiecaiMetr_auroc=valid_metrics.auroc
meanPiecaiMetr_AP=valid_metrics.AP
meanPiecaiMetr_score=valid_metrics.score
```
below image showing diffrent trials and recorded values auroc's are increasing with training but AP keep near 0 as both are more or less based on confusion matrix I have no idea how it can be possible
 

![image](https://user-images.githubusercontent.com/53857487/193086003-611dd72e-722d-4c45-ad23-fbbae26f516f.png)

**Comment by jakubMitura14** [comment]
Ok I have one more idea where I could made an error - I will investigate it before taking your time

**Comment by joeranbosma** [comment]
Hi @jakubMitura14,

One scenario in which I have had this before, is that I took the sigmoid/softmax twice before extracting the lesion candidates, resulting in lesion candidates of the whole image (because all values end up being rather close).
Best to plot the softmax values and generated detection maps for an easy case (for example case nr 32 of patient nr 32).

If neither this nor your hunch work out, happy to help debug further. 

Kind regards,
Joeran

**Comment by jakubMitura14** [comment]
Thank you so much, for tips I will try it out  if I would meet you some day on some conference I need to present you some Huge bottle of alcohol!

**Comment by joeranbosma** [comment]
Happy to help, and glad you're so enthusiastic! ^_^
(I'll be at RSNA 2022 by the way ;)
